mycelo/loadbot/bot.go

```diff
@@ -60,19 +85,35 @@ func runBot(ctx context.Context, acc env.Account, sleepTime time.Duration, clien
 	transactor.Context = ctx
 
 	for {
-		err := waitFor(ctx, sleepTime)
-		if err != nil {
-			return err
-		}
-
+		txSentTime := time.Now()
 		recipient, value := nextTransfer()
 		tx, err := stableToken.TxObj(transactor, "transfer", recipient, value).Send()
 		if err != nil {
+			if err != context.Canceled {
+				fmt.Printf("Error sending transaction: %v\n", err)
+			}
 			return fmt.Errorf("Error sending transaction: %w", err)
 		}
-		fmt.Printf("cusd transfer generated: from: %s to: %s amount: %s\ttxhash: %s\n", acc.Address.Hex(), recipient.Hex(), value.String(), tx.Transaction.Hash().Hex())
+		// fmt.Printf("cusd transfer generated: from: %s to: %s amount: %s\ttxhash: %s\n", acc.Address.Hex(), recipient.Hex(), value.String(), tx.Transaction.Hash().Hex())
 
-		printJSON(tx)
+		// printJSON(tx)
+		_, err = tx.WaitMined(ctx)
```

Comment:
i 'm not fully reviewing the whole logic.... one idea i had for this is to have another goroutine that will fetch every block, and inspect the txs...  if you have a "sentTxs" and then you check the block's tx you can easily see when they are mined... and you do all of that in a single routine...

to learn about sentTxs, every tx produce can send them to a channel.

Reply:
I'm not really sure what the idea is. This is also setup to deal with the case that the TPS is set higher than the network can handle (say when the block gas limit is below what the desired TPS should be).